### PR TITLE
refactor(git): sort git aliases alphabetically

### LINKED ISSUE #9895: docs(git): Sort git aliases alphabetically by command
Hi everyone,

So here we go:

**Is your feature request related to a particular plugin or theme? If so, specify it.**
Git plugin alias documentation

**Is your feature request related to a problem? Please describe.**
I would call it a problem, it is not a bug. Just a documentation improvement.

**Describe the solution you'd like**
I would like to sort the [**git** aliases table](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/git#aliases) alphabetically by command. I believe it is much easier to consult that way.

**Describe alternatives you've considered**
At the moment I am using my own fork, already sorted, to consult the git alias list.

**Additional context**
I opened a PR a while ago about this, #9743, but it is outdated, and with conflict. I will solve the conflict and resubmit.

**Related Issues**
I was not able to find any related issue.

**Note:**
I am new to the developing world, and this is my first contribution to an open source project❗ 😄 
Please excuse me any misunderstanding.

@robbyrussell 
@mcornella
@apjanke
@fred-o

### PR DESCRIPTION
**Pull request by jmschp**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- README - Git Plugin documentation: orders alphabetically the list of aliases by command.
- README - Git Plugin documentation: orders alphabetically functions by name.
- git.plugin.zsh - Order the alias following the README.
- git.plugin.zsh - Order functions by group and by function name.

## Other comments:

Makes it easier to find the needed alias for a command, in the docs.

Fixes #9895
Closes #11439 

@robbyrussell
@mcornella
@apjanke
@fred-o

### CODE REVIEW

File: plugins/git/README.md
Comment (cmalard) [inline plugins/git/README.md:18]: you sure about the alphabeticall order? ^^ (grt before ggpnp before g 😮 )

Comment (jmschp) [reply plugins/git/README.md:18]: The alphabetical order is on the  **Command** column, not the **Alias**.

Comment (jmschp) [reply plugins/git/README.md:21]: @cmalard Now what!? Lets merge!? 😄

Comment (TheTechRobo) [reply plugins/git/README.md:21]: The maintainers have a huge backlog of PRs; that's why there hasn't been much activity.

If you _really_ want it merged, try asking on the discord server.

Comment (jmschp) [reply plugins/git/README.md:21]: @TheTechRobo thanks for the hint!
I will definitely try the Discord server!
I would love to merge it. This is my first contribution to an open source project!

Comment (TheTechRobo) [reply plugins/git/README.md:21]: Congrats on your first contribution!! :D 

Yeah, I do believe it'll be useful. Much easier to read!

**Review by TheTechRobo — APPROVED** [review]

File: plugins/git/README.md
Comment (jmschp) [reply plugins/git/README.md:21]: 🙌

**Comment by fmalcher** [comment]
This should be merged with care since it will break all other PRs that change the README table. Maybe apply the sorting once all other PRs are merged?

**Comment by TheTechRobo** [comment]
Yeah, good point.

**Comment by jmschp** [comment]
Hey @mcornella
Any news about this PR?

**Comment by jmschp** [comment]
@mcornella 

can we merge this!? 🙏

**Comment by carlosala** [comment]
Hi! Could you share the script used to sort it, or you did that by hand? Thanks!!!

**Comment by jmschp** [comment]
@carlosala 

It is not exactly a script, I did it with Google Sheets.
Copied it to Google shees

**Comment by jmschp** [comment]
@carlosala 

By the way, today is my birthday, if we could get this merged, that would be a pretty nit present! 😁

**Comment by carlosala** [comment]
Hi @jmschp! Let's continue with this! Could you try sorting the aliases the same way you sorted the README? After that we'd be ready to merge :)

File: plugins/git/README.md
Comment (carlosala) [inline plugins/git/README.md:15]: Could we move this ones manually to its proper place?

**Comment by jmschp** [comment]
> @carlosala
> 
> By the way, today is my birthday, if we could get this merged, that would be a pretty nit present! 😁

Yes, I can. I will try this weekend.

**Comment by jmschp** [comment]
> Hi @jmschp! Let's continue with this! Could you try sorting the aliases the same way you sorted the README? After that we'd be ready to merge :)

@carlosala Yes, I can. I will try this weekend.

File: plugins/git/README.md
Comment (jmschp) [reply plugins/git/README.md]: You mean this should be right after

```
| gunwip | git rev-list --max-count=1 --format="%s" HEAD &#124; grep -q "--wip--" && git reset HEAD~1 |
```
?

File: plugins/git/git.plugin.zsh
Comment (jmschp) [inline plugins/git/git.plugin.zsh:153]: @carlosala 

There are this cases that don't match the order in the README, what do you think, leave them here or force tehm to match the README?

File: plugins/git/git.plugin.zsh
Comment (jmschp) [inline plugins/git/git.plugin.zsh:231]: I changed the order of the quotation marks `" '` to `' "`. Better for Google Sheets.

File: plugins/git/git.plugin.zsh
Comment (carlosala) [reply plugins/git/git.plugin.zsh:153]: I'd say match them, what do you think?

File: plugins/git/git.plugin.zsh
Comment (carlosala) [reply plugins/git/git.plugin.zsh:231]: if it doesn't change the functionality (like there's some variable or something like that it should be fine 👍🏻

File: plugins/git/git.plugin.zsh
Comment (jmschp) [reply plugins/git/git.plugin.zsh:153]: Yes, match the README. It terms of order, the README becomes the source of truth.

Comment (carlosala) [reply plugins/git/git.plugin.zsh:153]: Let's go for it then 👌🏻

File: plugins/git/README.md
Comment (jmschp) [inline plugins/git/README.md:241]: Removed the code formatter, so it is the same style as the rest of the table.

By the was should I go for ordering the functions by name?

**Comment by jmschp** [comment]
@carlosala 

It's ready. Take a look at my comments

File: plugins/git/README.md
Comment (carlosala) [reply plugins/git/README.md:241]: If you want to do it now and then we merge +1

Comment (jmschp) [reply plugins/git/README.md:241]: Yes, I will do it.

**Comment by jmschp** [comment]
@carlosala 

done! Let me know what you think.

**Comment by carlosala** [comment]
Thanks Miguel! It was an amazing contribution! 👏🏻

**Pull request merged**
